src/Domain/Creators/Python/Interval.cpp

```diff
@@ -15,9 +15,19 @@ namespace py = pybind11;
 namespace domain::creators::py_bindings {
 void bind_interval(py::module& m) {  // NOLINT
   py::class_<Interval, DomainCreator<1>>(m, "Interval")
-      .def(py::init<std::array<double, 1>, std::array<double, 1>,
-                    std::array<size_t, 1>, std::array<size_t, 1>,
-                    std::array<bool, 1>>(),
+      .def(py::init(
+               [](std::array<double, 1> lower_x, std::array<double, 1> upper_x,
+                  std::array<bool, 1> is_periodic_in_x,
```

Comment:
please move the `is_periodic_in_x` arg to the end, or move the `py::arg("is_periodic_in_x")` below to the correct position.